core/src/main/java/org/elasticsearch/index/mapper/geo/GeoShapeFieldMapper.java

```diff
@@ -257,31 +206,104 @@ public MappedFieldType clone() {
             return new GeoShapeFieldType(this);
         }
 
-        public PrefixTreeStrategy defaultStrategy() {
-            return this.defaultStrategy;
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof GeoShapeFieldType)) return false;
+            if (!super.equals(o)) return false;
+            GeoShapeFieldType that = (GeoShapeFieldType) o;
+            return Objects.equals(treeLevels, that.treeLevels) &&
+                Objects.equals(precisionInMeters, that.precisionInMeters) &&
+                Objects.equals(defaultDistanceErrorPct, that.defaultDistanceErrorPct) &&
+                Objects.equals(tree, that.tree) &&
+                Objects.equals(strategyName, that.strategyName) &&
+                Objects.equals(distanceErrorPct, that.distanceErrorPct) &&
+                Objects.equals(orientation, that.orientation);
         }
 
-        public PrefixTreeStrategy resolveStrategy(String strategyName) {
-            if (SpatialStrategy.RECURSIVE.getStrategyName().equals(strategyName)) {
-                return recursiveStrategy;
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), tree, strategyName, treeLevels, precisionInMeters, distanceErrorPct, defaultDistanceErrorPct, orientation);
+        }
+
+        @Override
+        public void freeze() {
```

Comment:
Is it ok for freeze() to do more than just freezing the state? In that case it looks used as a builder to me? (Maybe it's fine, I'm just wondering: will this fieldtype still work ok if you don't call freeze()?)